src/features/swaps/SwapScreen.tsx

```diff
@@ -225,15 +302,25 @@ const SwapScreen = () => {
     })
     if (!decision) return
 
-    submitTreasurySwap(new PublicKey(Mints[youPayTokenType]), youPayTokenAmount)
+    if (youPayTokenType === Tokens.HNT) {
+      submitMintDataCredits(youPayTokenAmount)
+    }
+
+    if (youPayTokenType !== Tokens.HNT) {
+      submitTreasurySwap(
+        new PublicKey(Mints[youPayTokenType]),
+        youPayTokenAmount,
+      )
+    }
 
-    navigation.push('SwappingScreen', {
+    navigation.replace('SwappingScreen', {
```

Comment:
Hmm not sure if this was intended. Will have to test this on the simulator

Reply:
Done!